iroh-base/src/node_addr.rs

```diff
@@ -6,31 +14,54 @@ use url::Url;
 
 use crate::key::{NodeId, PublicKey};
 
-/// A peer and it's addressing information.
+/// Network-level addressing information for an iroh-net node.
+///
+/// This combines a node's identifier with network-level addressing information of how to
+/// contact the node.
+///
+/// To establish a network connection to a node both the [`NodeId`] and one or more network
+/// paths are needed.  The network paths can come from various sources:
+///
+/// - A [discovery] service which can provide routing information for a given [`NodeId`].
+///
+/// - A [`RelayUrl`] of the node's [home relay], this allows establishing the connection via
+///   the Relay server and is very reliable.
+///
+/// - One or more *direct addresses* on which the node might be reachable.  Depending on the
+///   network location of both nodes it might not be possible to establish a direct
+///   connection without the help of a [Relay server].
+///
+/// This structure will always contain the required [`NodeId`] and will contain an optional
+/// number of network-level addressing information.  It is a generic addressing type used
+/// whenever a connection to other nodes needs to be established.
+///
+/// [discovery]: https://docs.rs/iroh_net/*/iroh_net/index.html#node-discovery
+/// [home relay]: https://docs.rs/iroh_net/*/iroh_net/relay/index.html
+/// [Relay server]: https://docs.rs/iroh_net/*/iroh_net/index.html#relay-servers
```

Comment:
As far as I'm aware this type of links are not verified when checking docs so, while these make sense, it would be good to avoid them in general